programs/account-compression/src/instructions/insert_into_indexed_array.rs

```diff
@@ -0,0 +1,165 @@
+use std::collections::HashMap;
+
+use aligned_sized::aligned_sized;
+use anchor_lang::{prelude::*, solana_program::pubkey::Pubkey};
+use ark_ff::BigInteger256;
+use ark_serialize::CanonicalDeserialize;
+use light_hasher::Poseidon;
+use light_indexed_merkle_tree::array::IndexingArray;
+
+use crate::RegisteredProgram;
+
+#[derive(Accounts)]
+pub struct InsertIntoIndexedArrays<'info> {
+    /// CHECK: should only be accessed by a registered program/owner/delegate.
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    pub registered_verifier_pda: Option<Account<'info, RegisteredProgram>>, // nullifiers are sent in remaining accounts. @ErrorCode::InvalidVerifier
+}
+
+/// Inserts every element into the indexed array.
+/// Throws an error if the element already exists.
+/// Expects an indexed queue account as for every index as remaining account.
+pub fn process_insert_into_indexed_arrays<'a, 'info>(
+    ctx: Context<'a, '_, '_, 'info, InsertIntoIndexedArrays<'info>>,
+    elements: &'a [[u8; 32]],
+    low_element_indexes: &'a [u16],
```

Comment:
yes right now low_element_indexes is not used it is a place holder.
The reason why I don't pass the actual low_element_index is twofold:
1. I don't have a way to get it in ts yet
2. we don't have onchain logic to correct the low nullifier index yet.

I am not sur we can search for the low nullifier onchain in production because I am not sure we can do 2800 comparisons. It might be too expensive.
@vadorovsky this would be something interesting to benchmark. If it's not possible we will need to extend the indexed array to be a concurrent indexed array.